fastly/data_source_datacenters.go

```diff
@@ -0,0 +1,101 @@
+package fastly
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"strconv"
+
+	gofastly "github.com/fastly/go-fastly/v6/fastly"
+	"github.com/fastly/terraform-provider-fastly/fastly/hashcode"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceFastlyDatacenters() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceFastlyDatacentersRead,
+
+		Schema: map[string]*schema.Schema{
+			"pops": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "A list of all Fastly POPs.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"code": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "A code representing the POP location.",
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The name of the POP.",
+						},
+						"group": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "A code representing the general region of the world in which the POP location resides.",
+						},
+						"shield": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "A code representing the shielding name of the POP. The value may be empty if the POP is not available for shielding.",
+						},
```

Comment:
I appreciate `coordinates` probably isn't that useful in practice, but for completeness sake is it worth including? 🤔 I'm leaning against _not_ including it, but thought I'd just ask quickly out of interest.

Reply:
This was my personal decision to not include it as I can't really see the exact use case for that, especially when it comes to the IaC. For now, I don't think we would need to include it. We can always come back to this later if someone asks, and it won't be a breaking change.